File: Sorting/Selection Sort/.ipynb_checkpoints/Selection-Sort-checkpoint.c
#include <stdio.h>
#include <stdlib.h>
#define SIZE 5

int findSmallest(int *arr){
    
    int smallest = arr[0];

    int smallest_index = 0;
    
    for (int i = 1; i < SIZE; i++) {
        
        if (arr[i] < smallest) {
            smallest = arr[i];
            smallest_index = i;
        }
    }
    
    return smallest_index;

}

int *selectionSort(int *arr){

    int *newArr = (int *)malloc(SIZE * sizeof(int));
    
    for (int i = 0; i < SIZE; i++) {
        
        int smallest = findSmallest(arr);
        newArr[i] = arr[smallest];

        arr[smallest] = INT_MAX;
    
    }
    
    return newArr;
}

int main(void) {

    int arr[SIZE] = {5, 3, 6, 2, 10};
    int *sortarr = selectionSort(arr);
    
    for (int i = 0; i < SIZE; i++) {
        printf("%d ", sortarr[i]);
    }
    
    return 0;

}

File: Sorting/Selection Sort/.ipynb_checkpoints/Selection-Sort-checkpoint.py
def find_smallest(list):
    
    smallest = list[0]
    smallest_index = 0
    
    for i in range(1, len(list)):
        if list[i] < smallest:
            smallest = list[i]
            smallest_index = i
    
    return smallest_index

def find_biggest(list):
    
    biggest = list[0]
    biggest_index = 0
    
    for i in range(1, len(list)):
        if list[i] > biggest:
            biggest = list[i]
            biggest_index = i
    
    return biggest_index

def selection_sort(list, ascending = True):
    
    new_list = []
    
    if ascending == True:
    
        for i in range(len(list)):
            new_list.append(list.pop(find_smallest(list)))
            
    elif ascending == False:
        
        for i in range(len(list)):
            new_list.append(list.pop(find_biggest(list)))
        
    return new_list

selection_sort([5, 3, 6, 2, 10], True)

selection_sort([5, 3, 6, 2, 10], False)